### PR TITLE
Add default module root setting for configuration

### PR DESCRIPTION
**Pull request by ryfu-msft**
Changes:
- Adds a new settings object for `configureBehavior`
- Adds a new field for `defaultModuleRoot` that determines the default directory where modules are installed to.

Tests:
- Verify settings value is parsed correctly
- Added E2E test to check that the module is installed to the directory specified from the settings.json file.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/4974)

### CODE REVIEW

**Review by JohnMcPMS — APPROVED** [review]

File: src/AppInstallerCLIE2ETests/ErrorCommand.cs [hunk 1]
```diff
@@ -14,6 +14,15 @@ namespace AppInstallerCLIE2ETests
     /// </summary>
     public class ErrorCommand
     {
+        /// <summary>
+        /// Reset settings file to avoid affecting output from error command.
```
Comment (florelis) [inline src/AppInstallerCLIE2ETests/ErrorCommand.cs:18]: Why did you add this? I'm guessing changing the settings in the Configure tests was affecting this? If it's that, maybe we should also add it as a OneTimeTearDown to these tests, so that it does not affect other tests if the order ever changes. But also keep it here if these tests expect clean settings

**Review by florelis — APPROVED** [review]

File: src/AppInstallerCLIE2ETests/ErrorCommand.cs (hunk 1 above)
Comment (ryfu-msft) [reply src/AppInstallerCLIE2ETests/ErrorCommand.cs:18]: Yup that's right. One of the tests in the error command checks if there are 2 lines returned from the error command but if the settings file is invalid, it will return >2. I think this should be sufficient since the error command only outputs to console. It doesn't change the settings file during the tests so it won't impact other tests.

**Review by florelis — APPROVED** [review]

**Review by florelis — APPROVED** [review]
